### PR TITLE
IB: Option to enable Relaxed Ordering

### PR DESCRIPTION
**Pull request by bureddy**

### CODE REVIEW

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI status: <b>PASS</b#  > (click for details)</summary>

**Note:** the artefacts will be deleted after 14-Mar-2020

Artefact file | Status
-- | --
[build_nccl_rdma_sharp_plugins.log](https://gist.githubusercontent.com/mellanox-github/97ad645df5d0bb42bd5fff2c222af815/raw/5debbbe2eadfcbfcead40fba2e3772ca4795560f/build_nccl_rdma_sharp_plugins.log) | :heavy_check_mark: PASS
[configure_sharp_startup.log](https://gist.githubusercontent.com/mellanox-github/97ad645df5d0bb42bd5fff2c222af815/raw/0dccaa66712d7b87fb282d9a4c06e56656b81afa/configure_sharp_startup.log) | :heavy_check_mark: PASS
[configure_sharp_stop.log](https://gist.githubusercontent.com/mellanox-github/97ad645df5d0bb42bd5fff2c222af815/raw/efb7b91dd0cbc849f2fdea803beb13d8cc923f18/configure_sharp_stop.log) | :heavy_check_mark: PASS
[run_nccl_test.log](https://gist.githubusercontent.com/mellanox-github/97ad645df5d0bb42bd5fff2c222af815/raw/a1ff5cdedce3bc2ad2785b1ad21646db1f26368a/run_nccl_test.log) | :heavy_check_mark: PASS

**Comment by mellanox-github** [comment]
Mellanox CI PASSed.

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI status: <b>UNKNOWN</b> (click for details)</summary>

**Note:** the artefacts will be deleted after 14-Mar-2020

Artefact file | Status
-- | --
[build_nccl_rdma_sharp_plugins.log](https://gist.githubusercontent.com/mellanox-github/846b4d445925d1f3968d3eb1de1c810c/raw/bd70e5439b6aae05c39f888ef8c0e441bcdf03cb/build_nccl_rdma_sharp_plugins.log) | :heavy_check_mark: PASS
[configure_sharp_startup.log](https://gist.githubusercontent.com/mellanox-github/846b4d445925d1f3968d3eb1de1c810c/raw/7ef8631791d4f5fa95438d675b315d5425031508/configure_sharp_startup.log) | :heavy_check_mark: PASS
[run_nccl_test.log](https://gist.githubusercontent.com/mellanox-github/846b4d445925d1f3968d3eb1de1c810c/raw/6e6cdb9cfd8c431cd46cea3623bfea87a0a211a3/run_nccl_test.log) | :question: UNKNOWN

**Comment by mellanox-github** [comment]
Mellanox CI FAILed.

**Comment by artemry-mlnx** [comment]
bot:mlx:test

**Comment by artemry-mlnx** [comment]
NCCL-SHARP plugin CI failed due to some issues with NCCL nightly. Restarted the CI.

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI status: <b>PASS</b#  > (click for details)</summary>

**Note:** the artefacts will be deleted after 15-Mar-2020

Artefact file | Status
-- | --
[build_nccl_rdma_sharp_plugins.log](https://gist.githubusercontent.com/mellanox-github/246c044978529f5a2c35c761217c3396/raw/14e765e6ae29feb3a6cd52083e57cb100caa8513/build_nccl_rdma_sharp_plugins.log) | :heavy_check_mark: PASS
[configure_sharp_startup.log](https://gist.githubusercontent.com/mellanox-github/246c044978529f5a2c35c761217c3396/raw/b0ab795fa1bb0e2c4b706cb9d0cf943453d5a4eb/configure_sharp_startup.log) | :heavy_check_mark: PASS
[configure_sharp_stop.log](https://gist.githubusercontent.com/mellanox-github/246c044978529f5a2c35c761217c3396/raw/a3e6b6f309d9fa9099113a339ac0c2d68ab48071/configure_sharp_stop.log) | :heavy_check_mark: PASS
[run_nccl_test.log](https://gist.githubusercontent.com/mellanox-github/246c044978529f5a2c35c761217c3396/raw/2164957a65d3d56a4a1ba83271fa473cb07dc9fe/run_nccl_test.log) | :heavy_check_mark: PASS

**Comment by mellanox-github** [comment]
Mellanox CI PASSed.

File: src/plugin.c [hunk 1]
```diff
@@ -151,6 +152,11 @@ ncclResult_t ncclIbInit(ncclDebugLogger_t logFunction) {
   if (ncclNIbDevs == -1) {
     pthread_mutex_lock(&ncclIbLock);
     wrap_ibv_fork_init();
+
+    if (ncclParamIbPciRelaxedOrdering() & !IBV_ACCESS_RELAXED_ORDERING) {
```
Comment (Sergei-Lebedev) [inline src/plugin.c]: is it supposed to be logical AND here? (not binary)

Comment (bureddy) [reply src/plugin.c]: i think binary would work. but logical is correct. fixed

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI status: <b>PASS</b#  > (click for details)</summary>

**Note:** the artefacts will be deleted after 16-Mar-2020

Artefact file | Status
-- | --
[build_nccl_rdma_sharp_plugins.log](https://gist.githubusercontent.com/mellanox-github/82742c2f2d4cd5c92550112c8a4e19d0/raw/63c2d003e6de1821f743f40504ad9d2ef58f87ee/build_nccl_rdma_sharp_plugins.log) | :heavy_check_mark: PASS
[configure_sharp_startup.log](https://gist.githubusercontent.com/mellanox-github/82742c2f2d4cd5c92550112c8a4e19d0/raw/bb6eb6c33193cf5d971a55cc44b206c232888302/configure_sharp_startup.log) | :heavy_check_mark: PASS
[configure_sharp_stop.log](https://gist.githubusercontent.com/mellanox-github/82742c2f2d4cd5c92550112c8a4e19d0/raw/f44d5ba9d7775af1a1f4ff5e84ce18de08410e44/configure_sharp_stop.log) | :heavy_check_mark: PASS
[run_nccl_test.log](https://gist.githubusercontent.com/mellanox-github/82742c2f2d4cd5c92550112c8a4e19d0/raw/768008457fa54ae295099ee7c2fa132b571fb822/run_nccl_test.log) | :heavy_check_mark: PASS

**Comment by mellanox-github** [comment]
Mellanox CI PASSed.

**Pull request merged**